src/pkg/backup/details/details_test.go

```diff
@@ -173,6 +173,544 @@ func (suite *DetailsUnitSuite) TestDetailsEntry_HeadersValues() {
+
+	return DetailsEntry{
+		RepoRef:     rr.String(),
+		ShortRef:    rr.ShortRef(),
+		ParentRef:   rr.ToBuilder().Dir().ShortRef(),
+		LocationRef: loc.String(),
+		ItemInfo:    info,
+	}
+}
+
+func (suite *DetailsUnitSuite) TestDetailsAdd_NoLocationFolders() {
+	t := suite.T()
+	table := []struct {
+		name  string
+		entry DetailsEntry
+		// shortRefEqual allows checking that OneDrive and SharePoint have their
+		// ShortRef updated in the returned entry.
+		//
+		// TODO(ashmrtn): Remove this when we don't need extra munging for
+		// OneDrive/SharePoint file name changes.
+		shortRefEqual assert.ComparisonAssertionFunc
+	}{
+		{
+			name:          "Exchange Email",
+			entry:         exchangeEntry(t, "foo", 42, ExchangeMail),
+			shortRefEqual: assert.Equal,
+		},
+		{
+			name:          "OneDrive File",
+			entry:         oneDriveishEntry(t, "foo", 42, OneDriveItem),
+			shortRefEqual: assert.NotEqual,
+		},
+		{
+			name:          "SharePoint File",
+			entry:         oneDriveishEntry(t, "foo", 42, SharePointLibrary),
+			shortRefEqual: assert.NotEqual,
+		},
+		{
+			name: "Legacy SharePoint File",
+			entry: func() DetailsEntry {
+				res := oneDriveishEntry(t, "foo", 42, SharePointLibrary)
+				res.SharePoint.ItemType = OneDriveItem
+
+				return res
+			}(),
+			shortRefEqual: assert.NotEqual,
+		},
+	}
+
+	for _, test := range table {
+		for _, updated := range []bool{false, true} {
+			suite.Run(fmt.Sprintf("%s Updated %v", test.name, updated), func() {
+				t := suite.T()
+
+				rr, err := path.FromDataLayerPath(test.entry.RepoRef, true)
+				require.NoError(t, err, clues.ToCore(err))
+
+				db := &Builder{}
+
+				// Make a local copy so we can modify it.
+				localItem := test.entry
+
+				err = db.Add(rr, &path.Builder{}, updated, localItem.ItemInfo)
+				require.NoError(t, err, clues.ToCore(err))
+
+				// Clear LocationRef that's automatically populated since we passed an
+				// empty builder above.
+				localItem.LocationRef = ""
+				localItem.Updated = updated
+
+				expectedShortRef := localItem.ShortRef
+				localItem.ShortRef = ""
+
+				deets := db.Details()
+				assert.Len(t, deets.Entries, 1)
+
+				got := deets.Entries[0]
+				gotShortRef := got.ShortRef
+				got.ShortRef = ""
+
+				assert.Equal(t, localItem, got, "DetailsEntry")
+				test.shortRefEqual(t, expectedShortRef, gotShortRef, "ShortRef")
+			})
+		}
+	}
+}
+
+func (suite *DetailsUnitSuite) TestDetailsAdd_LocationFolders() {
+	t := suite.T()
+
+	exchange1 := exchangeEntry(t, "foo1", 42, ExchangeMail)
+	oneDrive1 := oneDriveishEntry(t, "foo1", 42, OneDriveItem)
+	sharePoint1 := oneDriveishEntry(t, "foo1", 42, SharePointLibrary)
+	sharePointLegacy1 := oneDriveishEntry(t, "foo1", 42, SharePointLibrary)
+	sharePointLegacy1.SharePoint.ItemType = OneDriveItem
+
+	time.Sleep(1)
+
+	exchange2 := exchangeEntry(t, "foo2", 43, ExchangeMail)
+	exchange3 := exchangeEntry(t, "foo2", 43, ExchangeContact)
```

Comment:
For ease of reading and quick comparison, it would be nice for multi-typed services like exchange to contain type refs in the var names. 
```suggestion
	exchMail2 := exchangeEntry(t, "foo2", 43, ExchangeMail)
	exchContact1 := exchangeEntry(t, "foo2", 43, ExchangeContact)
```